shell/platform/darwin/ios/framework/Source/SemanticsObject.mm

```diff
@@ -517,6 +523,48 @@ - (NSString*)accessibilityLabel {
   return label;
 }
 
+- (bool)semanticsObjectContainsPoint:(SemanticsObject*)semanticsObject withPoint:(CGPoint)point {
+  const SkRect& rect = [semanticsObject node].rect;
+  CGRect localRect = CGRectMake(rect.x(), rect.y(), rect.width(), rect.height());
+  CGRect globalRect = ConvertRectToGlobal(semanticsObject, localRect);
+  return CGRectContainsPoint(globalRect, point);
+}
+
+- (double)getSize:(SemanticsObject*)semanticsObject {
+  const SkRect& rect = [semanticsObject node].rect;
+  return rect.width() * rect.height();
+}
+
+- (SemanticsObject*)search:(SemanticsObject*)semanticsObject withPoint:(CGPoint)point {
+  if ([semanticsObject children].count == 0) {
+    if ([self semanticsObjectContainsPoint:semanticsObject withPoint:point] &&
+        [self isFocusable:semanticsObject]) {
+      return semanticsObject;
+    }
+    return nil;
+  }
+
+  SemanticsObject* bestChild = nil;
+  for (SemanticsObject* child in [semanticsObject children]) {
+    if ([self semanticsObjectContainsPoint:child withPoint:point] &&
+        (bestChild == nil || [self getSize:child] < [self getSize:bestChild])) {
+      bestChild = child;
+    }
+  }
+  if (bestChild != nil) {
+    return [self search:bestChild withPoint:point];
+  }
+  if ([self semanticsObjectContainsPoint:semanticsObject withPoint:point] &&
+      [self isFocusable:semanticsObject]) {
+    return semanticsObject;
+  }
+  return nil;
+}
+
+- (id)_accessibilityHitTest:(CGPoint)point withEvent:(UIEvent*)event {
```

Comment:
Is this the private API that's being overwritten?  This needs a comment for sure.

Ideally we can detect that the OS stops using this private API and we will need to do something else.  @cyanglaz do we have any code that does this?  Would we want a test to fail, or... something?

Does this pass TestFlight?  Need to make sure Apple doesn't reject an app implementing this.

Reply:
> Ideally we can detect that the OS stops using this private API

I don't remember anything exactly like this. We need to figure out how to test accessibility first to let the OS trigger this method in a test.

> Does this pass TestFlight? Need to make sure Apple doesn't reject an app implementing this.

Yes, I was able to upload this engine with a dummy app code to TestFlight.

Reply:
Are there any other workarounds for this? I am pretty sure overriding a private API is against store policies. Policies whose application is spotty.

Reply:
The other workaround that we might be able to try is to use UIViews for all the semanticsObject and build the semantics tree as if it is a UIView tree.

Reply:
I usually test the submission by submit an app to testflight with the local engine, which passed with this PR. Do you know if there are other occasions that Apple would reject the app after passing testflight?
If we can be sure Apple won't reject this in the current iOS version, landing this workaround can fix the issue without customers waiting for a long term fix. (UIView approach or other approaches that we can come up with later)